shotgun_api3/shotgun.py

```diff
@@ -4470,6 +4470,28 @@ def _translate_filters_simple(sg_filter):
     if len(values) == 1 and isinstance(values[0], (list, tuple)):
         values = values[0]
 
+    # Payload optimization: Do not send a full object
+    # just send the `type` and `id` when combining related queries
+    disable_entity_optimization = os.environ.get(
+        "SHOTGUN_API_DISABLE_ENTITY_OPTIMIZATION"
+    )
+    if (
+        not disable_entity_optimization
+        and condition["path"] != "id"
+        and condition["relation"] in ["is", "is_not"]
+        and isinstance(values[0], dict)
+    ):
+        try:
+            values = [
+                {
+                    "type": values[0]["type"],
+                    "id": values[0]["id"],
+                }
```

Comment:
Why do we keep a dict?

I see in the description that we are reducing the payload to:
```python
sg.find('Asset', [['project', 'is', {'id': 72, 'type': 'Project'}]])
```

But why not to:
```python
sg.find('Asset', [['project', 'is', 72}]])
```

instead?

Reply:
That won't be accepted by SG

Reply:
Well I don't understand. This is exactly like this isn't it?
![image](https://github.com/user-attachments/assets/50981f53-384c-4ee7-9efa-c32f1b5871bf)

Reply:
It's not. Check the first argument of `find` or `find_one`. When entity `Project` is sent, you can pass directly the id as you want. When doing a relate query by sending an `Asset` entity, AND you want to query but asset's project ID, then you need to provide a project object dict.